code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,221 @@
+GLOBAL_LIST_EMPTY(virology_goals)
+GLOBAL_LIST_EMPTY(archived_virology_goals)
+
+/datum/virology_goal
+	var/name = "Generic Virology Goal"
+	/// The amount of units currently already delivered  
+	var/delivered_amount = 0  
+	/// The amount of units of the required virus that must be delivered for the completion of this goal  
+	var/delivery_goal = 15
+	var/completed = FALSE
+
+/datum/virology_goal/proc/get_report()
+	return "Complete this goal."
+
+/datum/virology_goal/proc/get_ui_report()
+	return "Complete this goal."
+
+/datum/virology_goal/proc/check_completion(list/datum/reagent/reagent_list)
+	check_total_virology_goals_completion()
+	return TRUE
+
+/datum/virology_goal/proc/check_for_duplicate()
+	return TRUE
+
+/proc/check_total_virology_goals_completion()
+	var/all_completed = TRUE
+	for(var/datum/virology_goal/V in GLOB.virology_goals)
+		if(!V.completed)
+			all_completed = FALSE
+			break
+	if(all_completed)
+		GLOB.archived_virology_goals += GLOB.virology_goals
+		GLOB.virology_goals = list(new/datum/virology_goal/propertysymptom, new/datum/virology_goal/virus, new/datum/virology_goal/virus/stealth)
+		for(var/obj/machinery/computer/pandemic/P in GLOB.pandemics)
+			P.print_goal_orders()
+
+/datum/virology_goal/Destroy()
+	LAZYREMOVE(GLOB.virology_goals, src)
+	. = ..()
+
+/datum/virology_goal/Topic(href, href_list)
+	..()
+
+	if(!check_rights(R_EVENT))
+		return
+
+	if(href_list["remove"])
+		qdel(src)
+
+/datum/virology_goal/propertysymptom
+	name = "Symptom With Properties Viral Sample Request"
+	var/goal_symptom //Type path of the symptom
+	var/goal_symptom_name
+	var/goal_property
+	var/goal_property_text
+	var/goal_property_value
+
+/datum/virology_goal/propertysymptom/New()
+	var/first_loop = TRUE
+	while(check_for_duplicate() || first_loop)
+		first_loop = FALSE
+		var/type = pick(subtypesof(/datum/symptom))
+		var/datum/symptom/S = new type()
+		goal_symptom = S.type
+		goal_symptom_name = S.name
+		goal_property = pick("resistance", "stealth", "stage_rate", "transmittable")
```

Comment:
Is there any specific reason it can't be like this? It would allow you to simplify out goal_property_text. I cant see any specific reason.
```suggestion
		goal_property = pick("resistance", "stealth", "stage rate", "transmittable")
```

Reply:
i could by renaming the property to "stage rate" everyone else in code which isnt a lot